packages/vendure-plugin-limit-variant-per-order/src/add-item-override.resolver.ts

```diff
@@ -70,12 +70,12 @@ export class AddItemOverrideResolver extends ShopOrderResolver {
     variantId: string | number,
     channelId: ID
   ): void {
-    const orderLine = order.lines.find(
-      (line) => line.productVariant.id == variantId
+    const orderLine = order.lines.find((line) =>
+      idsAreEqual(line.productVariant.id, variantId)
     )!;
```

Comment:
I propose we only iterate the order lines in the validate function, to prevent double iteration:

1. Refactor this part in the `adjustOrderLine` function:
```diff
-    const orderLine = order.lines.find((line) => line.id == args.orderLineId);
-     if (orderLine) {
-       this.validate(order, orderLine.productVariant.id, ctx.channelId);
-    }
+   this.validate(order, orderLine.productVariant.id, ctx.channelId);
```
2. In the `validate` function we remove the `!` check, and also approve if that orderline doesn't exists:
```diff
-  const orderLine = order.lines.find((line) =>  idsAreEqual(line.productVariant.id, variantId)!;
+ const orderLine = order.lines.find((line) =>  idsAreEqual(line.productVariant.id, variantId);
+ if (!orderLine) {
+     return; // No action needed, because the order doesn't have this orderLine anymore
+ }
```